src/nifake/tests/test_session.py

```diff
@@ -172,6 +114,110 @@ def test_get_string_attribute(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_ctypes_library.niFake_SetAttributeViReal64.side_effect = self.side_effects_helper.niFake_SetAttributeViReal64
+        self.side_effects_helper['SetAttributeViReal64']['return'] = test_error_code
+        self.patched_ctypes_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        with nifake.Session('dev1') as session:
+            try:
+                session.read_write_double = -42
+                assert False
+            except nifake.Error as e:
+                assert e.code == test_error_code
+                assert e.description == test_error_desc
+                self.patched_ctypes_library.niFake_SetAttributeViReal64.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', 1000001, -42)
+
+    def test_error_on_init(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_ctypes_library.niFake_InitWithOptions.side_effect = self.side_effects_helper.niFake_InitWithOptions
+        self.side_effects_helper['InitWithOptions']['return'] = test_error_code
+        self.side_effects_helper['InitWithOptions']['vi'] = SESSION_NUM_FOR_TEST
+        self.patched_ctypes_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        try:
+            nifake.Session('dev1')
+            assert False
+        except nifake.Error as e:
+            assert e.code == test_error_code
+            assert e.description == test_error_desc
+
+    def test_invalid_method_call_not_enough_parameters(self):
+        self.patched_ctypes_library.niFake_GetAStringWithSpecifiedMaximumSize.side_effect = self.side_effects_helper.niFake_GetAStringWithSpecifiedMaximumSize
+        with nifake.Session('dev1') as session:
+            try:
+                session.get_a_string_with_specified_maximum_size()
+                assert False
+            except TypeError as e:
+                pass
+
+    def test_invalid_method_call_wrong_type(self):
+        self.patched_ctypes_library.niFake_GetAStringWithSpecifiedMaximumSize.side_effect = self.side_effects_helper.niFake_GetAStringWithSpecifiedMaximumSize
+        with nifake.Session('dev1') as session:
+            try:
+                session.get_a_string_with_specified_maximum_size('potato')
+                assert False
+            except TypeError as e:
+                pass
+
+    def test_library_singleton(self):
+        with nifake.Session('dev1') as session:
+            lib1 = session.library
+        with nifake.Session('dev1') as session:
+            lib2 = session.library
+        assert lib1 == lib2
```

Comment:
lib1 is lib2 
(you don't want to compare, you want to ensure it's the same exact instance)